### PR TITLE
[1LP][RFR] Adds events API tests to Lenovo provider

### PR DESCRIPTION
**Pull request by Sanmoo**
Adds simple events API test to Lenovo provider

Depends on:
  * [ManageIQ/integration_tests](https://github.com/ManageIQ/integration_tests/pull/6689)

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6c602c43d68d65e56f15bc9c83e3cf5ec66ce449:

`cfme/common/provider_views.py`:
- :bangbang: Line 185:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 579:90: **E231** *missing whitespace after ','*
- :bangbang: Line 579:101: **E501** *line too long (106 > 100 characters)*

`cfme/physical/physical_server.py`:
- :bangbang: Line 196:1: **E302** *expected 2 blank lines, found 1*

`cfme/physical/provider/__init__.py`:
- :bangbang: Line 111:101: **E501** *line too long (101 > 100 characters)*

`cfme/tests/physical_infrastructure/test_physical_server.py`:
- :red_circle: Line 6:1: **F401** *'cfme.physical.physical_server.PhysicalServerCollection' imported but unused*
- :red_circle: Line 8:1: **F401** *'cfme.test_requirements' imported but unused*
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 18:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 23:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/physical_infrastructure/test_providers_events_api.py`:
- :bangbang: Line 10:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 17:1: **E302** *expected 2 blank lines, found 1*
- :red_circle: Line 19:29: **F821** *undefined name 'physical_server'*
- :heavy_exclamation_mark: Line 24:6: **W292** *no newline at end of file*

Hmmm
:scream: You might like to check the lint of your code :point_up: 

* 2c0ace705ced8769fb2c331615b5176ddd51feb6
  * Merge commit detected!!! :bangbang:
* 649c9ce618224b1058b1afde2e3022d97a2d8574
  * Merge commit detected!!! :bangbang:
* ab7ff95b59eabc1970033ed3d638f4f516f7d5cb
  * Merge commit detected!!! :bangbang:
* 70b40819ee08bd0503df08f788190b54b8d85f07
  * Merge commit detected!!! :bangbang:
* 05f14c325a12bad7ddf8ad79f9f6b0917599d83f
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ced6f83462770182fdad44ac48953fed665c1d75:

`cfme/common/provider_views.py`:
- :bangbang: Line 185:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 579:90: **E231** *missing whitespace after ','*
- :bangbang: Line 579:101: **E501** *line too long (106 > 100 characters)*

`cfme/physical/physical_server.py`:
- :bangbang: Line 196:1: **E302** *expected 2 blank lines, found 1*

`cfme/physical/provider/__init__.py`:
- :bangbang: Line 111:101: **E501** *line too long (101 > 100 characters)*

`cfme/tests/physical_infrastructure/test_physical_server.py`:
- :red_circle: Line 6:1: **F401** *'cfme.physical.physical_server.PhysicalServerCollection' imported but unused*
- :red_circle: Line 8:1: **F401** *'cfme.test_requirements' imported but unused*
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 18:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 23:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/physical_infrastructure/test_providers_events_api.py`:
- :bangbang: Line 10:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 17:1: **E302** *expected 2 blank lines, found 1*

Hmmm
:scream: You might like to check the lint of your code :point_up: 

* 2c0ace705ced8769fb2c331615b5176ddd51feb6
  * Merge commit detected!!! :bangbang:
* 649c9ce618224b1058b1afde2e3022d97a2d8574
  * Merge commit detected!!! :bangbang:
* ab7ff95b59eabc1970033ed3d638f4f516f7d5cb
  * Merge commit detected!!! :bangbang:
* 70b40819ee08bd0503df08f788190b54b8d85f07
  * Merge commit detected!!! :bangbang:
* 05f14c325a12bad7ddf8ad79f9f6b0917599d83f
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ae89aebd05a409ab7ef3bfb8beecf68831ca77bd:

`cfme/tests/physical_infrastructure/test_providers_events_api.py`:
- :bangbang: Line 10:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 17:1: **E302** *expected 2 blank lines, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by Sanmoo** [comment]
Hi,

The failures in the ci checks are not related to this pull request.

File: cfme/tests/physical_infrastructure/api/test_physical_server_events_api.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+DELAY = 60
+
+# All sources from all currently available physical infrastructure
+# providers
+SOURCES=['LenovoXclarity']
+
+pytest_generate_tests = testgen.generate([LenovoProvider], scope="class")
+
+
+@pytest.fixture(scope="class")
```
Comment (quarckster) [inline cfme/tests/physical_infrastructure/api/test_physical_server_events_api.py]: **REQUIRED**
The scope should be `module`

File: cfme/tests/physical_infrastructure/api/test_physical_server_events_api.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# The time interval to perform a verification of state change
+# in seconds (1 min)
+DELAY = 60
+
+# All sources from all currently available physical infrastructure
+# providers
+SOURCES=['LenovoXclarity']
+
+pytest_generate_tests = testgen.generate([LenovoProvider], scope="class")
```
Comment (quarckster) [inline cfme/tests/physical_infrastructure/api/test_physical_server_events_api.py]: **REQUIRED**
Please remove this line and change `pytestmark`:
```python
pytestmark = [
    pytest.mark.tier(3),
    pytest.mark.provider([LenovoProvider], scope="module")
]

Comment (Sanmoo) [reply cfme/tests/physical_infrastructure/api/test_physical_server_events_api.py]: Thanks @quarckster . I think I have addressed all your comments.

**Comment by dajoRH** [comment]
Lint report for commit 855699c501bbb9192eb1827a1956dc3d0787bfc7:

`cfme/tests/physical_infrastructure/api/test_physical_server_events_api.py`:
- :bangbang: Line 18:8: **E225** *missing whitespace around operator*
- :bangbang: Line 51:101: **E501** *line too long (112 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Pull request merged**